components/server/src/websocket/websocket-connection-manager.ts

```diff
@@ -279,19 +309,33 @@ class GitpodJsonRpcProxyFactory<T extends object> extends JsonRpcProxyFactory<T>
             throw new ResponseError(ErrorCodes.PERMISSION_DENIED, "not allowed");
         }
 
+        const span = opentracing.globalTracer().startSpan(method);
+        const ctx = { span };
+
+        // some generic data
+        const { clientId } = this.clientMetadata;
+        span.addTags({ clientId });
+        if (this.userId) {
+            span.addTags({ userId: this.userId });
```

Comment:
Unless we already use `userId` elsewhere I'd prefer `user.id` so we can add other user-related attributes under the `user.*` namespace.